external-lb.go

```diff
@@ -77,8 +76,8 @@ func updateExistingConfigs(metadataConfigs map[string]model.LBConfig, providerCo
 			mLBConfig := metadataConfigs[key]
 			pLBConfig := providerConfigs[key]
 			var update bool
-			//check that the targetName and targets match
-			if mLBConfig.LBTargetName == pLBConfig.LBTargetName {
+			//check that the targetPoolName and targets match
+			if mLBConfig.LBTargetPoolName == pLBConfig.LBTargetPoolName {
```

Comment:
Hold on, I think this check will cause following scenario to fail, so it should be removed.

If Service s1 uses a VirtualServer vs1 on f5
Then if I remove the label vs1 from s1 but add it to another service s2, this check will avoid updating f5